src/Tasks/Microsoft.NET.Build.Tasks/CheckForImplicitPackageReferenceOverrides.cs

```diff
@@ -22,7 +22,7 @@ public class CheckForImplicitPackageReferenceOverrides : TaskBase
 
         protected override void ExecuteCore()
         {
-            var duplicateItems = PackageReferenceItems.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1);
+            var duplicateItems = PackageReferenceItems.GroupBy(i => i.ItemSpec.ToLowerInvariant()).Where(g => g.Count() > 1);
```

Comment:
Use the GroupBy overload that takes a comparer and pass StringComparer.OrdinalIgnoreCase